### ISSUE TITLE
SSR nextjs query parameter receives the query strings with an extra set of quotes

### ISSUE THREAD

**Issue by divyanshu013** [appbaseio/reactivesearch#513]
**Issue Type:**

Enhancement

**Platform:**

Web

**Description:**

Currently we surround the `selectedValue` with double quotes `"`. This results in the `query` object in `getInitialProps` to include an extra set of quotes. If possible we can add a parser to ignore the extra set of quotes and check for possible edge cases.

**Screenshots:**

![image](https://user-images.githubusercontent.com/6682655/45164381-f2a5b600-b20f-11e8-837c-80b554eed540.png)

**Minimal reproduction of the problem with instructions:**

Try an app with nextjs using `URLParams` and check the `query` object in `getInitialProps`

**Reactivesearch version:** x.y.z

2.8.2

**Comment by axeff** [comment]
Just ran into the same problem.
In web/src/server/index.js initReactiveSearch fn:
`let value = getValue(searchState, label, component.defaultSelected);
if (value) {
    value = value.split('').map(item => item.replace('"', '')).join('');
}`
would do the trick.

**Comment by aheimlich** [comment]
The solution I found was to do the JSON parsing of the query string parameters inside your `getInitialProps` function.

```javascript
class Index extends React.Component {
  static async getInitialProps({ query }) {
    const searchState = {}

    Object.keys(query).forEach((key) => {
      try {
        const value = JSON.parse(query[key])
        searchState[key] = value
      } catch (e) {
        // Do not add to searchState if JSON parsing fails.
      }
    })

    const store = await initReactivesearch(
      componentCollection,
      searchState,
      settings
    )
    
    return {
      store
    }
  }
}
```

**Comment by metagrover** [comment]
Thanks, we will include this logic at the library level to improve the development experience.

**Comment by willopez** [comment]
on the other hand, this update breaks the SSR on sites that are not using NextJS. I am working with a server side rendered app with SSR and now the site does set the initial search state on the server. I see this error: `REACTIVESEARCH - could not parse the search state for componentId`

**Comment by willopez** [comment]
currently "working around" it by adding quotes to the search string 😕

**Comment by metagrover** [comment]
@willopez Fixed the parsing for non-next.js SSR apps. You will no longer need to add extra quotes to the search values. Fix will be out in `3.0.0-beta.8`.
